File: flink-jobs/src/main/java/ru/spb/itmo/pirsbd/asashina/flinkjobs/model/ClickEvent.java
package ru.spb.itmo.pirsbd.asashina.flinkjobs.model;

import java.time.LocalDateTime;

public record ClickEvent(String id, String type, LocalDateTime createdAt, LocalDateTime receivedAt, String sessionId,
                         String ip, Integer userId, String url, String referrer, String deviceType, String userAgent,
                         ClickEventPayload payload) {

}

File: generator/src/main/kotlin/ru/spb/itmo/pirsbd/asashina/generator/model/ClickEvent.kt
package ru.spb.itmo.pirsbd.asashina.generator.model

import java.time.LocalDateTime
import java.util.UUID

data class ClickEvent(
    var id: String = UUID.randomUUID().toString(),
    var type: String? = null,
    var createdAt: LocalDateTime? = null,
    var receivedAt: LocalDateTime? = null,
    var sessionId: String? = null,
    var ip: String? = null,
    var userId: Int? = null,
    var url: String? = null,
    var referrer: String? = null,
    var deviceType: String? = null,
    var userAgent: String? = null,
    var payload: ClickEventPayload? = null,
)